activesupport/lib/active_support/core_ext/string/output_safety.rb

```diff
@@ -37,7 +37,7 @@ def unwrapped_html_escape(s) # :nodoc:
       if s.html_safe?
         s
       else
-        s.gsub(HTML_ESCAPE_REGEXP, HTML_ESCAPE)
+        ActiveSupport::Multibyte::Unicode.tidy_bytes(s).gsub(HTML_ESCAPE_REGEXP, HTML_ESCAPE)
```

Comment:
@rafaelfranca - any thoughts on the above? Was hoping this would just be a quick fix!